src/app/report-user/report-user.component.ts

```diff
@@ -0,0 +1,105 @@
+import {Component, OnInit} from "@angular/core";
+import {ActivatedRoute, Router} from "@angular/router";
+import {HttpClient, HttpHeaders} from "@angular/common/http";
+import {AuthService} from "../service/auth.service";
+import {RecipeService} from "../service/recipe.service";
+import {ReportService} from "../service/report.service";
+import {DetailedRecipe} from "../model/detailed-recipe.model";
+import {ReportRequest} from "../model/report-request.model";
+
+@Component({
+  selector: 'report-user',
+  templateUrl: './report-user.component.html',
+  styleUrls: ['./report-user.component.css']
+})
+
+export class ReportUserComponent implements OnInit {
+  routeId: number;
+  report;
+  recipe: DetailedRecipe;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private http: HttpClient,
+    private authService: AuthService,
+    private recipeService: RecipeService,
+    private reportService: ReportService
+  ) {
+    this.report = new ReportRequest()
+  }
+
+  get isLoggedIn(): boolean {
+    return this.authService.authenticated;
+  }
+
+  ngOnInit() {
+    this.route.parent.url.subscribe(() => {
+      let path = this.route.snapshot.paramMap.get('previous');
+      this.routeId = parseInt(path.replace("/recipe/", ""));
+      this.report.reportingUserId = this.authService.getUserId()
```

Comment:
Actually I just noticed that the `getUserId` is broken in the `authService`. 
It always returns `undefined` because it never waits for the get to be completed. 

Because of this, the reporting feature does not work

Reply:
Dat is raar, bij mijn tests werkte het altijd. Dan is er een geval waarin het soms niet werkt. Ga ik morgen nalopen.

Reply:
Heb het gevonden, het kwam soms voor dat de request wel lukte maar het binden naar de variabele mislukte zonder error. Heb de tussenstap van de variabele eruit gehaald door een nieuwe getPrincipal methode toe te voegen aan de authservice en in de reportComponent forkJoin toegepast zodat de requests binnen dezelfde scope gebeuren.